scripts/__init__.py

```diff
@@ -29,9 +31,18 @@
 
 def run(prg_name):  # python 3.7+, otherwise define each script manually
     def fn():
-        res = subprocess.run(
-            [prg_name] + sys.argv[1:]
-        )  # run whatever you like based on 'name
-        if res.returncode:
-            sys.exit(res.returncode)
+        # run whatever you like based on 'name'
+        p = subprocess.Popen(
+            [prg_name] + sys.argv[1:])
+        try:
+            # check regularly p is alive
+            while p.poll() is None:
+                time.sleep(0.1) # time.sleep avoid too many CPU consumption.
```

Comment:
Why not working with signals?
https://stackabuse.com/handling-unix-signals-in-python/
http://avi.im/blag/2016/sigterm-in-python/

Reply:
You're right, I can use a signal to capture the keyboard interruption. And so stop the program.

But to catch all other signals? Should I list them all?

I need to check further more the solution, regarding signals.